devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -37,12 +37,15 @@ metadata {
 		fingerprint mfr:"003B", prod:"6341", model:"0544", deviceJoinName: "Schlage Camelot Touchscreen Deadbolt Door Lock"
 		fingerprint mfr:"003B", prod:"6341", model:"5044", deviceJoinName: "Schlage Century Touchscreen Deadbolt Door Lock"
 		fingerprint mfr:"003B", prod:"634B", model:"504C", deviceJoinName: "Schlage Connected Keypad Lever Door Lock"
-		fingerprint mfr:"0129", prod:"0002", model:"0800", deviceJoinName: "Yale Touchscreen Deadbolt Door Lock" // YRD120
-		fingerprint mfr:"0129", prod:"0002", model:"0000", deviceJoinName: "Yale Touchscreen Deadbolt Door Lock" // YRD220, YRD240
-		fingerprint mfr:"0129", prod:"0002", model:"FFFF", deviceJoinName: "Yale Touchscreen Lever Door Lock" // YRD220
-		fingerprint mfr:"0129", prod:"0004", model:"0800", deviceJoinName: "Yale Push Button Deadbolt Door Lock" // YRD110
-		fingerprint mfr:"0129", prod:"0004", model:"0000", deviceJoinName: "Yale Push Button Deadbolt Door Lock" // YRD210
-		fingerprint mfr:"0129", prod:"0001", model:"0000", deviceJoinName: "Yale Push Button Lever Door Lock" // YRD210
+		// Yale mfr could be both 0109 or 0129 for prod < 0005
+		fingerprint mfr:"0109", prod:"0001", deviceJoinName: "Yale Push Button Lever Door Lock" // YRD210
+		fingerprint mfr:"0129", prod:"0001", deviceJoinName: "Yale Push Button Lever Door Lock" // YRD210
+		fingerprint mfr:"0109", prod:"0002", deviceJoinName: "Yale Touchscreen Deadbolt Door Lock" // YRD120, YRD220, YRD240
+		fingerprint mfr:"0129", prod:"0002", deviceJoinName: "Yale Touchscreen Deadbolt Door Lock" // YRD120, YRD220, YRD240
+		fingerprint mfr:"0109", prod:"0002", deviceJoinName: "Yale Touchscreen Lever Door Lock" // YRD220
+		fingerprint mfr:"0129", prod:"0002", deviceJoinName: "Yale Touchscreen Lever Door Lock" // YRD220
+		fingerprint mfr:"0109", prod:"0004", deviceJoinName: "Yale Push Button Deadbolt Door Lock" // YRD110, YRD210
+		fingerprint mfr:"0129", prod:"0004", deviceJoinName: "Yale Push Button Deadbolt Door Lock" // YRD110, YRD210
 		fingerprint mfr:"0129", prod:"8002", model:"0600", deviceJoinName: "Yale Assure Lock with Bluetooth"
```

Comment:
That one in the ZWA doc that Duncan pointed you to is for the push button assure deadbolt, `prod 8004, model 0600`

It looks like all three of the touchscreen assure deadbolts listed have `prod 8002`. All are also listed to have `model 0600` regardless of having a keyway or not. They do not list any `prod 8002, model 1600`. Here's where I call foul. Clearly  the `model 1600` exists. I have one, we sell it, and its raw description is a few inches up the screen from here. Further, they claim that the YRD226-ZW2 is *both* an assure lock with a keyway and _without_ a keyway, both having `prod 8002` and `model 0600`. How does that make any sense?

- http://products.z-wavealliance.org/products/1973
- http://products.z-wavealliance.org/products/1878

Hint: the first link above is for the YRD246. The YRD226 has a keyway, the YRD246 does not. Another mistake in the ZWA data.

Look, the ZWA may have gotten erroneous information - I don't know. I'm guessing the data sheets sent by Yale make have been filled out quickly, with cut and paste errors or the like. In any case, I've seen several provable errors across different yale products at the ZWA product site that I've called out here. I don't trust it. 

After looking at this very closely now, I stand behind the changes I've requested. 
To capture all the Assure locks, use just the mfr and prod with the join names below. I don't know yet how to distinguish the bluetooth vs. the non-bluetooth touchscreen Assure locks, but this will get us reasonably correct in the meantime: 

`prod 8002` `Yale Assure Touchscreen Door Lock`
`prod 8004` `Yale Assure Push Button Door Lock`